src/schemapack/_internals/utils.py

```diff
@@ -93,3 +98,47 @@ def transient_directory_change(path: Path):
         yield
     finally:
         os.chdir(original_cwd)
+
+
+def model_to_serializable_dict(
+    model: BaseModel,
+) -> dict[str, Any]:
+    """Converts the provided pydantic model to a JSON-serializable dictionary.
+
+    Returns:
+        A dictionary representation of the provided model.
+    """
+    return json.loads(model.model_dump_json(exclude_defaults=True))
+
+
+def dumps_model(
+    model: BaseModel,
+    *,
+    yaml_format: bool = True,
+) -> str:
+    """Dumps the provided pydantic model as a JSON or YAML-formatted string.
+
+    Args:
+        model:
+            The model to dump.
+        yaml_format:
+            Whether to dump as YAML (`True`) or JSON (`False`).
+    """
+    model_dict = model_to_serializable_dict(model)
+
+    if yaml_format:
+        with StringIO() as buffer:
+            yaml.dump(model_dict, buffer)
+            return buffer.getvalue().strip()
+
+    return json.dumps(model_dict, indent=2)
+
+
+def write_dict(schemapack_dict: dict, *, path: Path, yaml_format: bool) -> None:
```

Comment:
The function name suggests this is a generic function, yet the argument `schemapack_dict` suggests it's not

Reply:
Right, I  renamed it generically.